### DISCUSSION TITLE
Propulsion engine data

### DISCUSSION THREAD

**Discussion by hudkinsnoah** [JSBSim-Team/jsbsim#1058] [Q&A]
Hello! I am continuing my work to find information for the T37 aircraft coming from the C310. I am looking at propulsion engine data and after printing the reference catalog, I see a lot of the data references that are no longer available with the T37 aircraft. Things such as `propulsion/engine[0 (or 1)]/oil-pressure-psi` and `propulsion/engine[0]/oil-temperature-degF`. I am also not seeing some references that I am looking for such as fuel pressure, torque, vacuum psi, and interstage turbine temperature(itt). I am not sure if any of these are available, but if they are, any help finding them would be appreciated. Thanks!

**Comment by seanmcleod** [comment]
You do realise that they have completely different engine types right? So there will be a number of properties that make sense for a piston engine but don't make sense for a jet engine and vice versa.

**Reply by hudkinsnoah** [reply]
Yes! I know that there are different engines in each of these planes so I know that there are going to be some that are missing. I guess I may not have clarified that very well. I still would think that there are some similarities such as oil pressure and oil temperature though. Fuel pressure, torque, vacuum psi, itt wouldn't make sense for the piston engine though (I think, my knowledge is only surface level of these).

**Comment by seanmcleod** [comment]
You can check what properties have been created over the years for the different engine types in the source code, and check how many are common.

https://github.com/JSBSim-Team/jsbsim/blob/b9bfc2350e0e00e53246b65979cc18df2922a88d/src/models/propulsion/FGPiston.cpp#L354-L392

https://github.com/JSBSim-Team/jsbsim/blob/b9bfc2350e0e00e53246b65979cc18df2922a88d/src/models/propulsion/FGTurbine.cpp#L562-L597

**Reply by hudkinsnoah** [reply]
This has certainly been helpful, thank you! I think that I have everything that I need (still checking other data refs I might need), but I am running into trying to calculate ITT in turbine engines. There doesn't appear to be a data reference for it and looking at other aircraft and engine types(turbo prop) it is available, but has quite a bit of calculations going on. Is the best way to get this into turbine to just simply copy all of the code over in order to get that, or is there another way I should calculate that?

The other thing, if I was to add the DHC6 aircraft, it is missing an n2 reference. Is there a way to get this calculation into turbo prop? It is kind of the same idea as the ITT calculation where there are a lot of calculations happening in the turbine engines that aren't in turbo prop which just means there would be a lot of copying and I am not sure that it would be a perfect or correct translation.

Thanks!
